### PR TITLE
Ignore sample-controller binary

### PR DESCRIPTION
**Pull request by Jefftree**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Add gitignore for sample-controller binary. Avoids accidental commits.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/assign @sttts
/cc @jpbetz

### CODE REVIEW

File: staging/src/k8s.io/sample-controller/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+sample-controller
```
Comment (sttts) [inline staging/src/k8s.io/sample-controller/.gitignore:1]: do we have the same for all binaries?

Comment (Jefftree) [reply staging/src/k8s.io/sample-controller/.gitignore:1]: We do have examples of binaries in the .gitignore:

- https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/pod-security-admission/.gitignore
- https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/.gitignore
- https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/mounter/.gitignore

but I'm sure there are more directories that produce binaries that do not have the ignore. One step at a time

Comment (sttts) [reply staging/src/k8s.io/sample-controller/.gitignore:1]: oh, you are saying we naturally produce that binary (during e2e?)

**Pull request merged**